airbyte-cdk/python/airbyte_cdk/sources/file_based/error_handlers/file_based_discover_error_handler.py

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+import logging
+from typing import Optional
+
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError, InvalidSchemaError, SchemaInferenceError
+from airbyte_cdk.sources.utils.discover_error_handler import AbstractDiscoverErrorHandler
+
+
+class FileBasedDiscoverErrorHandler(AbstractDiscoverErrorHandler):
+    def handle_discover_error(self, logger: logging.Logger, exception: Exception) -> Optional[Exception]:
+        """
+        Default File Based source implementation of the discover error handler. Logs the error or raises the exception.
+        """
+        exceptions_to_log = (InvalidSchemaError, SchemaInferenceError, ConfigValidationError)
```

Comment:
nit: 

We can make the class more generic by passing the types of exceptions to log in the constructor

```
class LogExpectedErrorsHandler(AbstractDiscoverErrorHandler):
    def __init__(exceptions_to_log: List[...]):
       self._exceptions_to_log = exceptions_to_log
   ...
```

Then in the filebased source:
```
    def discover_error_handler(self) -> AbstractDiscoverErrorHandler:
        return FileBasedDiscoverErrorHandler((InvalidSchemaError, SchemaInferenceError, ConfigValidationError))
```